File: frontend/src/lib/api/recommend.ts
import request from './request';

export interface SimilarDocumentItem {
	doc_id: number;
	title: string;
	original_file_name: string;
	summary: string;
	authors: string[];
	tags: string[];
	similarity: number;
}

export interface RecommendSimilarResponse {
	results: SimilarDocumentItem[];
}

export interface PersonalizedRecommendationsResponse {
	results: SimilarDocumentItem[];
}

export interface CollaboratorItem {
	user_id: number;
	username: string;
	nickname: string;
	avatar_url: string | null;
	similarity: number;
	signal_count: number;
}

export interface CollaboratorRecommendationsResponse {
	results: CollaboratorItem[];
}

const recommendApi = {
	getSimilar(
		docId: number,
		opts: { lab_id?: number; limit?: number } = {}
	): Promise<RecommendSimilarResponse> {
		const params: Record<string, string | number> = {};
		if (opts.lab_id != null) params.lab_id = opts.lab_id;
		if (opts.limit != null) params.limit = opts.limit;
		return request.get<RecommendSimilarResponse>(`/docs/${docId}/similar`, {
			params
		}) as unknown as Promise<RecommendSimilarResponse>;
	},
	getForUser(
		opts: { lab_id?: number; limit?: number } = {}
	): Promise<PersonalizedRecommendationsResponse> {
		const params: Record<string, string | number> = {};
		if (opts.lab_id != null) params.lab_id = opts.lab_id;
		if (opts.limit != null) params.limit = opts.limit;
		return request.get<PersonalizedRecommendationsResponse>('/mine/recommendations', {
			params: params,
			timeout: 30000
		}) as unknown as Promise<PersonalizedRecommendationsResponse>;
	},
	getCollaborators(opts: {
		lab_id: number;
		limit?: number;
	}): Promise<CollaboratorRecommendationsResponse> {
		const params: Record<string, string | number> = { lab_id: opts.lab_id };
		if (opts.limit != null) params.limit = opts.limit;
		return request.get<CollaboratorRecommendationsResponse>('/mine/collaborators', {
			params,
			timeout: 30000
		}) as unknown as Promise<CollaboratorRecommendationsResponse>;
	}
};

export default recommendApi;

File: svc-gateway/internal/dto/recommend.go
package dto

type RecommendSimilarQuery struct {
	LabID uint `form:"lab_id"`
	Limit int  `form:"limit" binding:"omitempty,min=1,max=20"`
}

type SimilarDocumentItem struct {
	DocID            uint     `json:"doc_id"`
	Title            string   `json:"title"`
	OriginalFileName string   `json:"original_file_name"`
	Summary          string   `json:"summary"`
	Authors          []string `json:"authors"`
	Tags             []string `json:"tags"`
	Similarity       float64  `json:"similarity"`
}

type RecommendSimilarResponse struct {
	Results []SimilarDocumentItem `json:"results"`
}

type RecommendForUserQuery struct {
	LabID uint `form:"lab_id"`
	Limit int  `form:"limit" binding:"omitempty,min=1,max=50"`
}

type RecommendForUserResponse struct {
	Results []SimilarDocumentItem `json:"results"`
}

type RecommendCollaboratorsQuery struct {
	LabID uint `form:"lab_id" binding:"required,gt=0"`
	Limit int  `form:"limit" binding:"omitempty,min=1,max=50"`
}

type CollaboratorItem struct {
	UserID      uint    `json:"user_id"`
	Username    string  `json:"username"`
	Nickname    string  `json:"nickname"`
	AvatarURL   *string `json:"avatar_url"`
	Similarity  float64 `json:"similarity"`
	SignalCount uint32  `json:"signal_count"`
}

type RecommendCollaboratorsResponse struct {
	Results []CollaboratorItem `json:"results"`
}
